packages/dev/inspector-v2/src/components/properties/particles/attractor.tsx

```diff
@@ -0,0 +1,117 @@
+    const impostor = CreateSphere("Attractor impostor #" + id, { diameter: 1 }, scene);
+    impostor.scaling.setAll(initialScale);
+    impostor.position.copyFrom(attractor.position);
+    impostor.material = initialMaterial;
+    impostor.reservedDataStore = { hidden: true };
+    return impostor;
+};
+
+async function CreateTextRendererAsync(scene: Scene, impostor: AbstractMesh, index: number, color: Color3) {
+    const sdfFontDefinition = await (await fetch("https://assets.babylonjs.com/fonts/roboto-regular.json")).text();
+    const fontAsset = new FontAsset(sdfFontDefinition, "https://assets.babylonjs.com/fonts/roboto-regular.png");
+
+    const textRenderer = await TextRenderer.CreateTextRendererAsync(fontAsset, scene.getEngine());
+    textRenderer.addParagraph("#" + index, {}, Matrix.Scaling(0.5, 0.5, 0.5).multiply(Matrix.Translation(0, 1, 0)));
+    textRenderer.isBillboard = true;
+    textRenderer.color = Color4.FromColor3(color, 1.0);
+    textRenderer.render(scene.getViewMatrix(), scene.getProjectionMatrix());
+    textRenderer.parent = impostor;
+    return textRenderer;
+}
+
+/**
+ * Represents the UX of an attractor, a sphere with a color/size whose position matches that of the underlying attractor
+ * @param props
+ * @returns
+ */
+export const AttractorComponent: FunctionComponent<AttractorProps> = (props) => {
+    const { attractor, id, impostorScale, impostorMaterial, scene } = props;
+    const classes = useAttractorStyles();
+    const [shown, setShown] = useState(true);
+
+    // Create observer and cleanup on unmount (we can't use useResource since Observer is not an IDisposable)
+    const sceneOnAfterRenderObserverRef = useRef<Observer<Scene>>();
+    useEffect(() => () => sceneOnAfterRenderObserverRef.current?.remove(), []);
+
+    // We only want to recreate the impostor mesh and associated if id, scene, or attractor/impostor changes
+    const impostor = useResource(useCallback(() => CreateImpostor(id, scene, attractor, impostorScale, impostorMaterial), [id, scene, attractor]));
+    const label = useAsyncResource(useCallback(async () => await CreateTextRendererAsync(scene, impostor, id, props.impostorColor), [scene, impostor, id]));
+
+    // If impostor, color, or label change, recreate the observer function so that it isnt hooked to old state
+    useEffect(() => {
+        sceneOnAfterRenderObserverRef.current?.remove();
+        sceneOnAfterRenderObserverRef.current = scene.onAfterRenderObservable.add(() => {
+            attractor.position.copyFrom(impostor.position);
+            if (label) {
+                label.color = Color4.FromColor3(props.impostorColor);
+                label.render(scene.getViewMatrix(), scene.getProjectionMatrix());
+            }
+        });
+    }, [impostor, label, props.impostorColor]);
+
+    useEffect(() => {
+        impostor.scaling.setAll(impostorScale);
+    }, [impostorScale]);

```

Comment:
`imposter` should be the dep list too, right? As the code is written, the imposter instance would change if id, scene, or attractor changed.